src/processor/operator/scan_node_id.cpp

```diff
@@ -98,5 +106,14 @@ void ScanNodeID::setSelVector(ExecutionContext* context, NodeTableScanState* tab
         context->clientContext->getTx(), outValueVector);
 }
 
+double ScanNodeID::getProgress(ExecutionContext* context) const {
+    if (!sharedState->getCurrentTableState()) {
```

Comment:
As we discussed, let's keep `numNodes` and `numNodesScanned` in shared state. `numNodes` should be updated once only in `initGlobalStateInternal` and `numNodesScanned` is updated each getMorsel is called.